build.gradle.kts

```diff
@@ -0,0 +1,340 @@
+import com.android.build.gradle.BaseExtension
+import com.jfrog.bintray.gradle.BintrayExtension
+import com.jfrog.bintray.gradle.RecordingCopyTask
+import org.jfrog.gradle.plugin.artifactory.dsl.ArtifactoryPluginConvention
+import org.jfrog.gradle.plugin.artifactory.dsl.DoubleDelegateWrapper
+import org.jfrog.gradle.plugin.artifactory.dsl.PublisherConfig
+import org.jfrog.gradle.plugin.artifactory.task.ArtifactoryTask
+
+buildscript {
+    repositories {
+        google()
+        jcenter()
+        mavenLocal()
+    }
+
+    dependencies {
+        classpath(Config.Plugins.android)
+        classpath(Config.Plugins.kotlin)
+        classpath(Config.Plugins.google)
+        classpath(Config.Plugins.bintray)
+        classpath(Config.Plugins.buildInfo)
+    }
+}
+
+// See https://github.com/gradle/kotlin-dsl/issues/607#issuecomment-375687119
+subprojects { parent!!.path.takeIf { it != rootProject.path }?.let { evaluationDependsOn(it) } }
+
+allprojects {
+    repositories {
+        google()
+        jcenter()
+        mavenLocal()
+    }
+
+    // Skip Javadoc generation for Java 1.8 as it breaks build
+    if (JavaVersion.current().isJava8Compatible) {
+        tasks.withType<Javadoc> {
+            options {
+                this as StandardJavadocDocletOptions
+                addStringOption("Xdoclint:none", "-quiet")
+            }
+        }
+    }
+
+    if ((group as String).isNotEmpty() && name != "lint" && name != "internal") {
+        configureAndroid()
+        configureQuality()
+
+        val isLibrary = name == "library"
+        if (Config.submodules.contains(name) || isLibrary) {
+            setupPublishing(isLibrary)
+        }
+    }
+}
+
+val Project.configDir get() = "$rootDir/library/quality"
+val Project.reportsDir get() = "$buildDir/reports"
+
+fun Project.configureAndroid() {
+    apply(plugin = "com.android.${if (name == "app" || name == "proguard-tests") {
```

Comment:
This is over my personal fanciness limit :-)

How about:

```
if (name == "app" || name == "proguard-tests") {
  // apply app plugin
} else {
  // apply library plugin
```

Reply:
Haha, np. 😁😊